### PR TITLE
Removed X-content-length header check in favour of json key. Switched…

### PR DESCRIPTION
**Pull request by pringelmann**
… to using urls as PK's for relations

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -87,33 +93,31 @@ export const convertRESTRequestToHTTP = ({
...
     }
 };
 
+/**
+ * Returns the last param of a given url
+ * @param {string} url
+ */
+const extractPrimaryKey = url => {
+    let substrs = url.slice('/');
+    return substrs[substrs.length - 2];
```
Comment (jwfwessels) [inline src/index.js]: um shouldn't this be -1 🤔

File: src/index.js [hunk 2]
```diff
@@ -135,7 +139,7 @@ export default (apiUrl, httpClient = fetchJson) => {
      * @returns {Promise} the Promise for a REST response
      */
 
-    return (type, resource, params) => {
+    return async (type, resource, params) => {
```
Comment (jwfwessels) [inline src/index.js:142]: why async?

File: src/index.js [hunk 3]
```diff
@@ -145,16 +149,8 @@ export default (apiUrl, httpClient = fetchJson) => {
...
-                convertHTTPResponseToREST({
-                    response,
-                    type,
-                    resource,
-                    params
-                })
-            );
-            return Promise.all(
-                url.filter((_, i) => i > 0).map(singleUrl =>
+            const responses = await Promise.all(
```
Comment (jwfwessels) [inline src/index.js:152]: not quite following you can explain on a call maybe.

**Review by jwfwessels — APPROVED** [review]
looks mostly fine just not following one or two parts.

File: src/index.js (hunk 2 above)
Comment (pringelmann) [reply src/index.js:142]: cleaner syntax

File: src/index.js (hunk 3 above)
Comment (pringelmann) [reply src/index.js:152]: same as previous implementation, just using async/await instead of promise return

Comment (jwfwessels) [reply src/index.js:152]: i understand now

File: src/index.js (hunk 2 above)
Comment (jwfwessels) [reply src/index.js:142]: i understand now

**Pull request merged**